InteractiveHtmlBom/web/render.js

```diff
@@ -632,6 +640,28 @@ function pointWithinDistanceToSegment(x, y, x1, y1, x2, y2, d) {
   return dx * dx + dy * dy <= d * d;
 }
 
+function modulo(n, mod) {
+  return ((n % mod) + mod ) % mod;
+}
+
+function pointWithinDistanceToArc(x, y, xc, yc, radius, startangle, endangle, d) {
+  var angle1 = modulo(startangle, 360.0);
+  var angle2 = modulo(endangle, 360.0);
+  var dx = x - xc;
+  var dy = y - yc;
+  var r = Math.sqrt(dx * dx + dy * dy);
+  var dr = Math.abs(radius - r);
+
+  if ( dr > d )
+    return false;
+
+  var angle = modulo(Math.atan2(dy, dx) * 180 / Math.PI, 360.0);
+  if (angle1 > angle2)
+    return (angle > (angle2 - radius * d) || angle < (angle1 + radius * d));
```

Comment:
Math here is fishy. What is `radius * d` supposed to mean? I think `d` should not enter the equation here at all.

My approach would be
1. If distance from point to center differs from radius by more than d then return false (just like you have)
2. If distance from point to either arc endpoint is less than d return true
3. Otherwise check that [dx,dy] vector angle is within [startangle, endangle] (with normalization caveats like you did)

Reply:
`d / radius` was what I meant to use.  For d << radius, chord length is approximately equal to arc length which is `radius * theta` (chord_angle) so this was meant to effectively extend the angular region by a distance approximately equal to d on either end.  Not exact, just a shortcut.

I did that rather than checking euclidian distance because the arc object only contains a start point, so calculating an end point to work with was just one more set of calculations.

Reply:
> For d << radius, chord length is approximately equal to arc length

I'm pretty sure that's true for small angles, not for small distances. But even so, fat tracks with relatively small radiuses are to be expected and we can't make that assumption.

Why not just follow my algo? I did a crude sketch in freecad to illustrate it

![image](https://user-images.githubusercontent.com/3622532/109435353-f5dc2000-79ce-11eb-96bb-24ecc6567e28.png)

1. checks that the point is in the right band
2. checks for the end caps (A and B regions, it also has some overlap with C region which is inconsequential)
3. checks that we are in the right sector of the band (C region)

Reply:
Yeah, I understood what you meant.  

I was reminded that the "start" point of an arc is really the center point, so actually both of the arc endpoints needed to be calculated first in order to check Euclidian distance.  

Anyway, easy enough - more lines of code and 4 more trig operations, but no square roots.